### ISSUE TITLE
Adapting to use with DataTables Editor?

### ISSUE THREAD

**Issue by asmaloney** [jbox-web/ajax-datatables-rails#23]
Thanks for this gem Antonio.

I'm interested in using the DataTables Editor (https://editor.datatables.net/) and it would be nice to use one class for both pagination/display and editing of a model.  I was thinking this could be done by modifying the **AjaxDatatablesRails::Base** class to work with the JSON the editor uses:

  https://editor.datatables.net/manual/server

Am I on the right track?  Any thoughts/suggestions/insights?

**Comment by antillas21** [comment]
Hi Andy,

Glad you like this gem and are looking for a way to implement additional behavior.

Taking a look at the links you provided, and specially to the page that provides documentation on what JSON requests/responses should be exchanged.

I would suggest the following:
- Don't modify the `AjaxDatatablesRails::Base` class, but extend it instead :wink:  ... something like this could work:

``` ruby
class AjaxDatatablesRails::Editor < AjaxDatatablesRails::Base
  # define all overrides and extensions you will need to work with
  # and implement the dataTables Editor specific requirements.
end

# so, when it's time to use it, say for example on a Contacts example
class ContactsEditorDatatable < AjaxDatatablesRails::Editor
  # define the specific methods needed to interact
  # with the model(s) required in this datatable
end

# and in your controller, you can have a common interface
render json: { ContactsEditorDatatable.new(view_context, *other_args) }
```
- Doing this, will provide you common methods that the base class provides and you may require in this specific use case.
- In other words: **Don't hack core**

Most important of all, extending the base class into a common stock class that can be used with the Editor, will allow you to contribute back to this project :smile: 

I would like to see what you get to build for this particular case. Let me know if I can help in any way.

**Comment by asmaloney** [comment]
Thanks Antonio.  If we end up going with DataTables + Editor this will be helpful.  And if I come up with something useful I'll be sure to feed it back in :grin:

**Comment by antillas21** [comment]
Great!

Let me know how it goes. I would like very much to see what you come up with.
- Antonio

**Comment by asmaloney** [comment]
@antillas21 It turns out we have decided not to use Datatables at the moment, so I won't have anything to report back...

**Comment by antillas21** [comment]
Ok @asmaloney ... I am closing this issue then.
